frontend/app/services/lookup-service.server.ts

```diff
@@ -593,21 +586,46 @@ function createLookupService() {
     const url = `${INTEROP_API_BASE_URI}/lookups/marital-statuses`;
     const response = await fetch(url);
 
-    const maritalStatusSchemaList = z.array(maritalStatusSchema);
+    if (!response.ok) {
+      log.error('%j', {
+        message: 'Failed to fetch data',
+        status: response.status,
+        statusText: response.statusText,
+        url: url,
+        responseBody: await response.text(),
+      });
 
-    if (response.ok) {
-      return maritalStatusSchemaList.parse(await response.json());
+      throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
     }
 
-    log.error('%j', {
-      message: 'Failed to fetch data',
-      status: response.status,
-      statusText: response.statusText,
-      url: url,
-      responseBody: await response.text(),
+    const maritalStatusSchema = z.object({
+      value: z.array(
+        z.object({
+          OptionSet: z.object({
+            Options: z.array(
+              z.object({
+                Value: z.number(),
+                Label: z.object({
+                  LocalizedLabels: z.array(
+                    z.object({
+                      Label: z.string(),
+                    }),
+                  ),
+                }),
+              }),
+            ),
+          }),
+        }),
+      ),
     });
 
-    throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
+    const parsedMaritalStatusData = maritalStatusSchema.parse(await response.json());
+    return parsedMaritalStatusData.value[0].OptionSet.Options.map((o) => ({
+      id: o.Value,
+      code: o.Value,
+      nameEn: o.Label.LocalizedLabels[0].Label,
+      nameFr: o.Label.LocalizedLabels[1].Label,
```

Comment:
FYI this can be done in a future PR since when we switch over to an actual API call, we won't know the order in which the labels are returned to us. 

We should be using `1033` for English and `1036` for French. For reference: https://github.com/DTS-STN/canadian-dental-care-plan/blob/948f4bb3f458cc39c64c56bf4f0ead344b832c53/frontend/app/services/letters-service.server.ts#L72

(I didn't know that `1033`/`1036` would be reused so `ENGLISH_LETTER_LANGUAGE_CODE` and `FRENCH_LETTER_LANGUAGE_CODE` should eventually be renamed to handle marital status lookups too)

Reply:
agreed and corrected